crates/tx3-lang/src/ast.rs

```diff
@@ -699,11 +699,18 @@ impl VariantCase {
     }
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub enum AssetName {
+    Ascii(String),
+    String(StringLiteral),
+    HexString(HexStringLiteral),
+}
+
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
 pub struct AssetDef {
     pub name: String,
     pub policy: Option<HexStringLiteral>,
-    pub asset_name: Option<String>,
+    pub asset_name: Option<AssetName>,
     pub span: Span,
 }
```

Comment:
I'm afraid that the Asset definition is becoming too opinionated and very coupled to a specific use-case.
I prefer if we turn into something that looks like this...

```pest
asset_def = {
    "asset" ~ identifier ~ "=" ~ data_expr ~ "." ~ data_expr ~ ";"
}
```

The analysis phase should be the one throwing an diagnostic error if any of the data_expr can't be coerced into bytes.